approval_polls/static/approval_polls/create.js

```diff
@@ -1,33 +1,41 @@
-$(function () {
-  var that = this;
-  this.numChoiceFields = 4;
+$(function() {
+  var that = this; 
+  this.numChoiceFields = 4
+  this.currentNum = undefined;
```

Comment:
"Data path" is perhaps a bad term. Specifically I was referring to how you broke up `this.numChoiceFields` into `this.numChoiceFields`, `this.currentNum` and the parameter `numChoiceFields`. 

Looking at it now I sort of see what you were doing. I think `this.NumCoiceFields` can be removed safely. It just sits there holding it's `4` and is never read or written to that I can see. 

`#choices-count` is named counter intuitively, Something like `#LastId` would be better. It might be a good idea to rename the javascript variable used to determine the next id(`this.currentNum`) into `this.LastId`, since Id numbers no longer have much to do with the number of choices. It's optional though.